internal/service/alarms/internal/serviceconfig/serviceconfig.go

```diff
@@ -0,0 +1,235 @@
+package serviceconfig
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strconv"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	api "github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/internal/db/models"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients/k8s"
+	serverutils "github.com/openshift-kni/oran-o2ims/internal/service/common/utils"
+	"github.com/stephenafamo/bob/dialect/psql"
+	"github.com/stephenafamo/bob/dialect/psql/dm"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	cleanupCronJobName   = "alarms-server-events-cleanup"
+	cleanupConfigMapName = "alarms-server-events-cleanup-sql"
+	cleanScriptDir       = "/scripts"
+	cleanScriptName      = "cleanup.sql"
+)
+
+var (
+	cronSchedule          = "0 * * * *" // run at the start of every hour e.g 2:00, 3:00 etc
+	jobBackoffLimit int32 = 3
+	successLimit    int32 = 3
+	failureLimit    int32 = 1
+	commonLabels          = map[string]string{
+		"app": utils.InventoryAlarmServerName,
+	}
+)
+
+// Config defines the configuration for serviceconfig
+type Config struct {
+	PostgresImage string `envconfig:"POSTGRES_IMAGE" required:"true"`
+	PodNamespace  string `envconfig:"POD_NAMESPACE" required:"true"`
+}
+
+func LoadConfig() (Config, error) {
+	var config Config
+	if err := envconfig.Process("", &config); err != nil {
+		return config, err //nolint:wrapcheck
+	}
+
+	return config, nil
+}
+
+// EnsureCleanupCronJob starts (or updates) a cronjob with all the required resources to do alarms events cleanup.
+func EnsureCleanupCronJob(ctx context.Context, sc *models.ServiceConfiguration, pg *pgxpool.Pool, hc client.Client) error {
+	config, err := LoadConfig()
```

Comment:
Do we want to do this every time or should we consider moving this `AlarmSeverConfig` and doing something similar to `config.LoadFromEnv()` which is run on startup?

Reply:
Yeah started looking into using AlarmSeverConfig at first...but ran into the circular import. Let me see if i can move it around a bit and pull this one time!

Reply:
should be done one time now at start now. Reusing AlarmSeverConfig required a bit too much refactoring so using unique struct still. Let me know how it looks!